src/Controls/src/Core/Handlers/Items/iOS/ItemsViewController.cs

```diff
@@ -577,7 +577,7 @@ internal void UpdateView(object view, DataTemplate viewTemplate, ref UIView uiVi
 
 				uiView?.Dispose();
 				uiView = null;
-
+				formsElement?.Handler?.DisconnectHandler();
```

Comment:
Do we need os on ItemsViewCOntroller2 ?

Reply:
@rmarinho ,
I've tested this scenario with CollectionViewHandler2, and since the header and footer handling differs in that implementation, the issue does not occur. Dynamically setting the header, footer, and empty view to null works without any exceptions in CollectionViewHandler2, so this fix isn't needed there.